packages/docusaurus/src/webpack/utils.ts

```diff
@@ -198,22 +198,24 @@ function filterWarnings(
   return warnings.filter((warning) => !isWarningFiltered(warning));
 }
 
-export function compile(config: Configuration[]): Promise<void> {
+export function compile(config: Configuration[]): Promise<Stats.ToJsonOutput> {
   return new Promise((resolve, reject) => {
     const compiler = webpack(config);
     compiler.run((err, stats) => {
       if (err) {
         reject(new Error(err.toString()));
       }
+      // let plugins consume all the stats
+      const allStats = stats?.toJson('errors-warnings');
       if (stats?.hasErrors()) {
-        stats.toJson('errors-only').errors.forEach((e) => {
+        allStats.errors.forEach((e) => {
           console.error(e);
         });
         reject(new Error('Failed to compile with errors.'));
       }
       if (stats?.hasWarnings()) {
         // Custom filtering warnings (see https://github.com/webpack/webpack/issues/7841).
-        let {warnings} = stats.toJson('errors-warnings');
+        let warnings = [...allStats.warnings];
```

Comment:
are there cases where the warning array is not defined?

Maybe we could replace the condition with `if (allStats.warnings?.length > 0)`, and TS would understand better that the array is there or something?

Reply:
This is there so the original `stats.warnings` doesn't get modified.

Reply:
I see thanks.
It shouldn't be, but won't harm to keep this code